apps/web-giddh/src/app/vat-report/obligations/obligations.component.ts

```diff
@@ -126,7 +135,42 @@ export class ObligationsComponent implements OnInit, OnDestroy {
             });
         } else {
             this.getFormControl('branchUniqueName').setValue(this.generalService.currentBranchUniqueName);
+            this.getCurrentCompanyBranchTaxNumber();
         }
+        this.taxNumber$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response?.body?.length) {
+                this.taxesList = response.body.map(tax => ({
+                    label: tax,
+                    value: tax
+                }));
+                if (this.taxesList.length === 1) {
+                    this.getFormControl('taxNumber').patchValue(this.taxesList[0].value);
+                }
+                if (this.isCompanyMode) {
+                    this.hasTaxNumber = true;
+                }
+                this.getVatObligations();
+            }
+        });
+    }
+
+    /**
+    * Get Current company branches information
+    *
+    * @private
+    * @memberof ObligationsComponent
+    */
+    private getCurrentCompanyBranchTaxNumber(): void {
+        this.componentStore.currentCompanyBranches$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                const currentBranchUniqueName = this.generalService.currentBranchUniqueName;
+                let currentBranch = cloneDeep(response.find(branch => branch?.uniqueName === currentBranchUniqueName));
```

Comment:
check without cloneDeep if work remove it

Reply:
Done